### ISSUE TITLE
Add body transform prop to handle body types other than json

### ISSUE THREAD

**Issue by tkh44** [tkh44/holen#1]

**Comment by vesparny** [comment]
Hi @tkh44. May I help on this one?

**Comment by tkh44** [comment]
Yes!

**Comment by vesparny** [comment]
What do you exactly have in mind?

**Comment by tkh44** [comment]
@vesparny I think I want to add a `responseTransformer` prop or something like that. The default would be:

```jsx
(res) => res.json()
```

The only tricky part is I think we also need to pass in onError and onSuccess handlers to this transformer so that we can hook it into the existing success/error handling.

Maybe something like

```jsx
(res, onSuccess, onError) => res.json().then(onSuccess).catch(onError)
```

What do you think?

**Comment by tkh44** [comment]
I also wonder if it should be more generic so users can do more than just transform the request. Maybe the more generic version can live on a different prop.

**Comment by vesparny** [comment]
At the moment `unfetch` supports : 
* text
* json
* xml
* blob

what about having an API like:

```jsx
<Holen 
  body={JSON.stringify({ message: 'hello' })}
  method="POST"
  url="api.startup.com/users"
  type="json"
  transformResponse={(data) => {
   data.isDecorated = true
   return data
 }}
>
  {({data}) => <pre>{data.isDecorated}</pre>}
</Holen>
```

**Comment by tkh44** [comment]
That looks great!

**Comment by vesparny** [comment]
Will work on it tomorrow :)

cheers

**Comment by tkh44** [comment]
closed in #3
